### PR TITLE
PROFIN-10

### PR DESCRIPTION
**Pull request by martinjulian**
Migracion para crear tabla 'services'

`create table services (id integer not null, days bigint, service_type varchar(255), sku varchar(255), primary key (id)) engine=InnoDB`

### CODE REVIEW

File: src/main/java/com/coderace/controller/ServiceController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            return ResponseEntity.status(HttpStatus.CREATED).body(responseDTO);
+        } catch (BadRequestException e) {
+            return ResponseEntity.status(e.getStatusCode()).body(e.getMessage());
+        }
+    }
+
+    @GetMapping
+    public ResponseEntity<Object> getAll() {
+        return ResponseEntity.ok().body(this.service.getAll());
+    }
```
Comment (code-race-2021) [inline src/main/java/com/coderace/controller/ServiceController.java:32]: menor, cambiaría ResponseEntity de Object por ResponseEntity de ServiceResponseDTO en ambos endpoint

File: src/main/java/com/coderace/model/enums/ServiceType.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.coderace.model.enums;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum ServiceType {
```
Comment (code-race-2021) [inline src/main/java/com/coderace/model/enums/ServiceType.java:6]: Agregar test unitario de esta clase, seguir el ejemplo de DeliveryType y DeliveryTypeTest

File: src/test/java/com/coderace/controller/ServiceControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+@WebMvcTest
+@DisplayName("ExampleController test | Unit")
+class ServiceControllerTest {
```
Comment (code-race-2021) [inline src/test/java/com/coderace/controller/ServiceControllerTest.java]: - cambiar DisplayName a ServiceController test | Unit
- Agregarle a la anotación @WebMvcTest -> (controllers = ServiceController.class)

File: src/main/java/com/coderace/model/entities/Service.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.coderace.model.enums.ServiceType;
+import lombok.Getter;
+
+import javax.persistence.*;
+
+@Entity
+@Getter
+@Table(name = "services")
+public class Service {
```
Comment (code-race-2021) [inline src/main/java/com/coderace/model/entities/Service.java:11]: Agregar test unitario 'hack' de esta clase, igual que Delivery y DeliveryTest

File: src/main/java/com/coderace/model/enums/ServiceType.java (hunk 2 above)
Comment (martinjulian) [reply src/main/java/com/coderace/model/enums/ServiceType.java:6]: [listo ](https://github.com/code-race-2021/final-project-cruds/commit/4ca4de77f632dc45026a93c680e29f5398d55b3c)

File: src/test/java/com/coderace/controller/ServiceControllerTest.java (hunk 3 above)
Comment (martinjulian) [reply src/test/java/com/coderace/controller/ServiceControllerTest.java]: [listo ](https://github.com/code-race-2021/final-project-cruds/commit/4ca4de77f632dc45026a93c680e29f5398d55b3c)

**Comment by code-race-2021** [comment]
# [Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) Report
> Merging [#7](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) (241fedb) into [master](https://codecov.io/gh/code-race-2021/final-project-cruds/commit/61fe2483a6dad16159e6d9ce4ca85eba59442c43?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) (61fe248) will **increase** coverage by `0.40%`.
> The diff coverage is `97.22%`.

[Impacted file tree graph](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)

```diff
@@             Coverage Diff              @@
##             master       #7      +/-   ##
============================================
+ Coverage     95.72%   96.12%   +0.40%     
- Complexity      124      164      +40     
============================================
  Files            20       26       +6     
  Lines           234      310      +76     
  Branches          3        4       +1     
============================================
+ Hits            224      298      +74     
  Misses            4        4              
- Partials          6        8       +2     
```


| [Impacted Files](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) | Coverage Δ | |
|---|---|---|
| [...ava/com/coderace/model/dtos/ServiceRequestDTO.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvbW9kZWwvZHRvcy9TZXJ2aWNlUmVxdWVzdERUTy5qYXZh) | `90.90% <90.90%> (ø)` | |
| [...va/com/coderace/model/dtos/ServiceResponseDTO.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvbW9kZWwvZHRvcy9TZXJ2aWNlUmVzcG9uc2VEVE8uamF2YQ==) | `90.90% <90.90%> (ø)` | |
| [...ava/com/coderace/controller/ServiceController.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvY29udHJvbGxlci9TZXJ2aWNlQ29udHJvbGxlci5qYXZh) | `100.00% <100.00%> (ø)` | |
| [...main/java/com/coderace/model/entities/Service.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvbW9kZWwvZW50aXRpZXMvU2VydmljZS5qYXZh) | `100.00% <100.00%> (ø)` | |
| [...ain/java/com/coderace/model/enums/ServiceType.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvbW9kZWwvZW51bXMvU2VydmljZVR5cGUuamF2YQ==) | `100.00% <100.00%> (ø)` | |
| [...main/java/com/coderace/service/ServiceService.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2Uvc2VydmljZS9TZXJ2aWNlU2VydmljZS5qYXZh) | `100.00% <100.00%> (ø)` | |
| [...ain/java/com/coderace/service/DeliveryService.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2Uvc2VydmljZS9EZWxpdmVyeVNlcnZpY2UuamF2YQ==) | `100.00% <0.00%> (ø)` | |
| [...a/com/coderace/model/dtos/DeliveryResponseDTO.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvbW9kZWwvZHRvcy9EZWxpdmVyeVJlc3BvbnNlRFRPLmphdmE=) | `90.90% <0.00%> (+3.40%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None). Last update [61fe248...241fedb](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/7?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

File: src/main/java/com/coderace/model/entities/Service.java (hunk 4 above)
Comment (martinjulian) [reply src/main/java/com/coderace/model/entities/Service.java:11]: [listo ](https://github.com/code-race-2021/final-project-cruds/commit/139e5c0105276bdb36ab240e3824163e2c32b342)

Comment (martinjulian) [reply src/main/java/com/coderace/model/entities/Service.java:11]: [listo ](https://github.com/code-race-2021/final-project-cruds/pull/7/commits/241fedb23cad14f813375c4f3e33d911ce011306)

File: src/main/java/com/coderace/controller/ServiceController.java (hunk 1 above)
Comment (martinjulian) [reply src/main/java/com/coderace/controller/ServiceController.java:32]: esto queda como igual entonces

**Review by NeiMeli — APPROVED** [review]
Excelente!!!

**Pull request merged**